### DISCUSSION TITLE
Change in when the expired objects are deleted

### DISCUSSION THREAD

**Discussion by balazsrunge** [minio/minio#13482] [Q&A] [answered]
Hi,

We just updated MINIO from https://github.com/minio/minio/releases/tag/RELEASE.2020-04-02T21-34-49Z to https://github.com/minio/minio/releases/tag/RELEASE.2021-04-22T15-44-28Z.

We had a lifecycle rule configured for a whole bucket (prefix is empty) with 1 day expiry time.

Using the older version objects were deleted after reaching the age of 24 hours. With the new version objects were deleted only at midnight after they are expired (So in worst case scenario the objects are deleted almost 1 day after they were expired.). 

I browsed all the change logs between these two version but I was not able to find the exact commit where this was modified.
Can you help me pinpoint the exact version where this was introduced?

Thanks and BR,
Balazs Runge

**Comment by klauspost** [comment]
@balazsrunge Upgrade to latest version and check.

**Reply by balazsrunge** [reply]
@klauspost: 
Thanks. Unfortunately it is not that easy to do the upgrade in our environment. Even if we would do it that would not help us, because we need to use that version for a while. We already had to do a lot of modifications due to the new version of MINIO. I just would like to know if this change in the behavior of lifecycle handling is well-known and will not be changed in future releases. But first I would like to know with what change it was introduced.

**Reply by klauspost** [reply]
Sorry, I cannot help you then. If you need support on older versions look into our [subscription offers](https://min.io/pricing).

**Reply by harshavardhana** [reply]
> @klauspost: 
> Thanks. Unfortunately it is not that easy to do the upgrade in our environment. Even if we would do it that would not help us, because we need to use that version for a while. We already had to do a lot of modifications due to the new version of MINIO. I just would like to know if this change in the behavior of lifecycle handling is well-known and will not be changed in future releases. But first I would like to know with what change it was introduced.

Lifecycle has went through a lot of changes so we won't know what you are really talking about.

What is the reason to stick to Apache version ? Is this something that you are trying to avoid using AGPLv3 versions?

**Reply by balazsrunge** [reply]
It was about when the objects marked for deletion because of expiration was carried out immediately or only at midnight.
But you already answered my question. 
Thanks, Balazs

**Comment by harshavardhana** [comment, marked as answer]
.
> 
> Using the older version objects were deleted after reaching the age of 24 hours. With the new version objects were deleted only at midnight after they are expired (So in worst case scenario the objects are deleted almost 1 day after they were expired.). 
> 

This behavior was changed to comply with AWS S3 implementation. Our earlier implementation was not correct.

**Reply by balazsrunge** [reply]
Thanks harshavardhana for the information!
I suspected that after reading the lifecycle management details in AWS S3 documentation, but I didn't know that the different behavior of lifecycle management in MINIO was intentional.
But If you say that the earlier configuration was incorrect then I guess this won't be changed in the future.

**Reply by klauspost** [reply]
No

**Reply by balazsrunge** [reply]
Thanks!
